languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml

```diff
@@ -41769,6 +41769,110 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
+                <example>Nasceu em Niterói–RJ.</example>
+            </antipattern>
+            <rule> <!-- #1: brackets -->
+                <pattern>
+                    <token case_sensitive="yes" regexp="yes">\p{Lu}\p{Ll}*</token>
+                    <marker>
+                        <token>(</token>
+                        <token case_sensitive="yes" regexp="yes">&state_abbrev_brazil;</token>
+                        <token>)</token>
+                    </marker>
+                </pattern>
+                <filter class="org.languagetool.rules.UnderlineSpacesFilter" args="underlineSpaces:before"/>
+                <message>Separe municípios e estados brasileiros com uma meia-risca.</message>
+                <suggestion>–\3</suggestion>
+                <example correction="–RJ">Nasceu em Niterói<marker> (RJ)</marker>.</example>
+                <example correction="–RJ">Nasceu em Niterói<marker>(RJ)</marker>.</example>
+                <example correction="–RJ">Nasceu em Niterói<marker> (RJ)</marker> e morreu lá.</example>
+            </rule>
+            <rule> <!-- #2: simple separators -->
+                <pattern>
+                    <token case_sensitive="yes" regexp="yes">\p{Lu}\p{Ll}*</token>
+                    <marker>
+                        <token/>
+                        <token case_sensitive="yes" regexp="yes">&state_abbrev_brazil;</token>
+                    </marker>
+                </pattern>
+                <filter class="org.languagetool.rules.UnderlineSpacesFilter" args="underlineSpaces:before"/>
+                <message>Separe municípios e estados brasileiros com uma meia-risca.</message>
+                <suggestion>–\3</suggestion>
+                <example correction="–RJ">Nasceu em Niterói<marker> – RJ</marker>.</example>
+                <example correction="–RJ">Nasceu em Niterói<marker>/RJ</marker>.</example>
+                <example correction="–RJ">Nasceu em Niterói<marker> — RJ</marker>.</example>
+                <example correction="–RJ">Nasceu em Niterói<marker>-RJ</marker>.</example>
+                <example correction="–RJ">Nasceu em Niterói<marker>, RJ</marker>.</example>
+            </rule>
+        </rulegroup>
+
+        <!--Some Brazilian states have the same abbreviation as US ones;
+            we'll prioritise the former, but this does create a couple of issues
+        e.g. MA is both Maine and Maranhão; Portland, MA will be corrected to Portland–MA...
+            Check AP #2 here for a naïve implementation. At some point, it might be a good idea to
+            actually check the *city name*, but this is a bigger thing.
+        -->
+        <!--State abbreviations common to both Brazil and the US:
+            AL: Alagoas/Alabama
+            MA: Maranhão/Maine
+            MS: Mato Grosso do Sul/Mississippi
+            MT: Mato Grosso/Montana
+            PA: Pará/Pennsylvania
+            SC: Santa Catarina/South Carolina
+        -->        
```

Comment:
I have compiled a list of cities in those Brazilian states, since those are the more urgent ones: [Municípios Brasil.zip](https://github.com/languagetool-org/languagetool/files/13661608/Municipios.Brasil.zip), maybe we could turn them into an entity?

Reply:
An entity would prob. be super slow. If we have a list, though, we could just add it to `resource/pt` and have a simple Java filter to perform a lookup, maybe?